dataplane/pkg/common/dataplane.go

```diff
@@ -211,7 +214,15 @@ func CreateDataplane(dp NSMDataplane, dataplaneProbes *DataplaneProbes) *Datapla
 	} else {
 		dataplaneProbes.SetSocketListenReady()
 	}
-	dataplane.RegisterDataplaneServer(config.GRPCserver, dp)
+
+	chain := ChainOf(dp,
+		DirectMemifInterfaces(config.NSMBaseDir),
+		Connect(utils.EnvVar("VPPAGENT_ENDPOINT").GetStringOrDefault("localhost:9111")),
```

Comment:
Presuming DirectMemifInterfaces() plus memif into the NSM Forwarder, and KernelInterface() plugs in kernel interfaces... why is Connect() before KernelInterfaces()?

Reply:
> why is Connect() before KernelInterfaces()?
Connect() makes connection to configurator server
`KernelInterface()` uses the this client